src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,75 +2,97 @@
+
+        productDao.setProductList(testProducts);
+
     }
 
     @Test
     public void testGetProduct() throws ProductNotFoundException {
-        Currency usd = Currency.getInstance("USD");
-        Product expectedProduct = new Product(3L, "sgs3", "Samsung Galaxy S III", new BigDecimal(300), usd, 5, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S%20III.jpg");
-        Product actualProduct = productDao.getProduct(3L);
-        assertEquals(expectedProduct, actualProduct);
+        Product actualProduct = productDao.getProduct(15L);
+        assertEquals(product1, actualProduct);
+
     }
 
     @Test(expected = ProductNotFoundException.class)
     public void testGetProductException() throws ProductNotFoundException {
-        Product actualProduct = productDao.getProduct(14L);
+        productDao.getProduct(14L);
+
     }
 
     @Test
-    public void testFindProductsWithNotNullPriceAndStockMoreThan0()
-    {
-        List<Product> products = productDao.findProducts();
-        int amountFalse = 0;
-        for (Product product: products) {
-            if (product.getPrice() == null || product.getStock() <= 0)
-                amountFalse++;
-        }
-        assertTrue(amountFalse == 0);
+    public void testFindProductsWithNotNullPriceAndStockMoreThan0() {
+        long amountFalse = productDao.findProducts()
+                .stream()
+                .filter(product -> product.getPrice() == null || product.getStock() <= 0)
+                .count();
+        assertEquals(0, amountFalse);
+
     }
 
     @Test
     public void testSaveNewOrExistingProduct() throws ProductNotFoundException {
-        Currency usd = Currency.getInstance("USD");
-        Product productToSave = new Product( "sgs_new", "New Product Test", new BigDecimal(300), usd, 5, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S%20III.jpg");
         productDao.save(productToSave);
         assertTrue(productToSave.getId() > 0);
         Product actualProduct = productDao.getProduct(productToSave.getId());
         assertEquals(productToSave, actualProduct);
 
-        productToSave = new Product(3L, "sgs_updated", "Updated Samsung Galaxy S III", new BigDecimal(300), usd, 5, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S%20III.jpg");
-        productDao.save(productToSave);
-        actualProduct = productDao.getProduct(productToSave.getId());
-        assertEquals(productToSave, actualProduct);
+        productDao.save(product1);
+        actualProduct = productDao.getProduct(product1.getId());
+        assertEquals(product1, actualProduct);
+
     }
 
     @Test
     public void testDeleteProduct() throws ProductNotFoundException {
-        long idToDelete = 13L;
+        long idToDelete = 15L;
+        int size = testProducts.size();
```

Comment:
Don't check by size, instead check that product is no more present in testProducts collection